python/mxnet/text/embedding.py

```diff
@@ -0,0 +1,600 @@
+                embedding_cls.pretrained_file_name_sha1.keys())
+                    for embedding_name, embedding_cls in
+                    registry.get_registry(TokenEmbedding).items()}
+
+
+@TokenEmbedding.register
+class GloVe(TokenEmbedding):
+    """The GloVe token embedding.
+
+    GloVe is an unsupervised learning algorithm for obtaining vector
+    representations for words. Training is performed on aggregated global
+    word-word co-occurrence statistics from a corpus, and the resulting
+    representations showcase interesting linear substructures of the word vector
+    space. (Source from https://nlp.stanford.edu/projects/glove/)
+
+    Reference:
+    GloVe: Global Vectors for Word Representation
+    Jeffrey Pennington, Richard Socher, and Christopher D. Manning
+    https://nlp.stanford.edu/pubs/glove.pdf
+
+    Website:
+    https://nlp.stanford.edu/projects/glove/
+
+    To get the updated URLs to the externally hosted pre-trained token embedding
+    files, visit https://nlp.stanford.edu/projects/glove/
+
+
+    Parameters
+    ----------
+    pretrain_file : str, default 'glove.840B.300d.txt'
+        The name of the pre-trained token embedding file.
+    embed_root : str, default '~/.mxnet/embeddings/'
+        The root directory for storing embedding-related files.
+    unknown_vec : callback
+        The callback used to initialize the embedding vector for the unknown
+        token.
+    """
+
+    # Map a pre-trained token embedding archive file and its SHA-1 hash.
+    pretrained_archive_name_sha1 = \
+        {'glove.42B.300d.zip': 'f8e722b39578f776927465b71b231bae2ae8776a',
+         'glove.6B.zip': 'b64e54f1877d2f735bdd000c1d7d771e25c7dfdc',
+         'glove.840B.300d.zip': '8084fbacc2dee3b1fd1ca4cc534cbfff3519ed0d',
+         'glove.twitter.27B.zip': 'dce69c404025a8312c323197347695e81fd529fc'}
+
+    # Map a pre-trained token embedding file and its SHA-1 hash.
+    pretrained_file_name_sha1 = \
+        {'glove.42B.300d.txt': '876767977d6bd4d947c0f84d44510677bc94612a',
+         'glove.6B.50d.txt': '21bf566a9d27f84d253e0cd4d4be9dcc07976a6d',
+         'glove.6B.100d.txt': '16b1dbfaf35476790bd9df40c83e2dfbd05312f1',
+         'glove.6B.200d.txt': '17d0355ddaa253e298ede39877d1be70f99d9148',
+         'glove.6B.300d.txt': '646443dd885090927f8215ecf7a677e9f703858d',
+         'glove.840B.300d.txt': '294b9f37fa64cce31f9ebb409c266fc379527708',
+         'glove.twitter.27B.25d.txt':
+             '767d80889d8c8a22ae7cd25e09d0650a6ff0a502',
+         'glove.twitter.27B.50d.txt':
+             '9585f4be97e286339bf0112d0d3aa7c15a3e864d',
+         'glove.twitter.27B.100d.txt':
+             '1bbeab8323c72332bd46ada0fc3c99f2faaa8ca8',
+         'glove.twitter.27B.200d.txt':
+             '7921c77a53aa5977b1d9ce3a7c4430cbd9d1207a'}
+
+    url_prefix = 'http://nlp.stanford.edu/data/'
+
+    def __init__(self, pretrained_file_name='glove.840B.300d.txt',
+                 embedding_root='~/.mxnet/embeddings/',
```

Comment:
This needs to be updated to `os.path.join` because that function takes care of OS-specific path separator. See #9352 and https://discuss.gluon.ai/t/topic/3041/49

Reply:
Resolved.